That's a weird build failure, especially as it worked in a local build:

```
[107/182] Generating bin/ldc2.o
FAILED: cd /home/vsts/work/1/s/parts/ldc-bootstrap/src && /usr/bin/ldmd2 -c -wi -O -inline -release -J/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd -J/home/vsts/work/1/s/parts/ldc-bootstrap/src/res -I/home/vsts/work/1/s/parts/ldc-bootstrap/src -I/home/vsts/work/1/s/parts/ldc-bootstrap/build -version=IN_LLVM -I/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd -of/home/vsts/work/1/s/parts/ldc-bootstrap/build/bin/ldc2.o /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/access.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/aggregate.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/aliasthis.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/apply.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/argtypes.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/arrayop.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/arraytypes.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/astcodegen.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/attrib.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/blockexit.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/builtin.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/canthrow.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/clone.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/compiler.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/complex.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/cond.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/console.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/constfold.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/cppmangle.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/cppmanglewin.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/ctfeexpr.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/ctorflow.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dcast.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dclass.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/declaration.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/delegatize.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/denum.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dimport.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dinterpret.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dmacro.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dmangle.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dmodule.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/doc.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dscope.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dstruct.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dsymbol.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dsymbolsem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dtemplate.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/dversion.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/entity.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/errors.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/escape.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/expression.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/expressionsem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/func.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/globals.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/gluelayer.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/hdrgen.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/hooks.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/id.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/identifier.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/impcnvtab.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/imphint.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/init.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/initsem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/inline.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/inlinecost.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/intrange.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/json.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/lambdacomp.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/ldcbindings.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/lexer.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/mars.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/mtype.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/nogc.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/nspace.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/objc.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/opover.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/optimize.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/parse.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/parsetimevisitor.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/permissivevisitor.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/printast.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/aav.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/array.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/ctfloat.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/file.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/filename.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/hash.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/longdouble.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/port.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/rmem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/rootobject.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/speller.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/stringtable.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/safe.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/sapply.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/semantic2.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/semantic3.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/sideeffect.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/statement.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/statement_rewrite_walker.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/statementsem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/staticassert.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/staticcond.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/target.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/templateparamsem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/tokens.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/traits.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/transitivevisitor.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/typesem.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/typinf.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/utf.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/utils.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/visitor.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/gen/dpragma.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/gen/ldctraits.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/gen/llvmhelpers.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/gen/logger.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/gen/semantic.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/driver/cache_pruning.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/driver/config.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/driver/configfile.d /home/vsts/work/1/s/parts/ldc-bootstrap/src/driver/main.d
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(310): Error: 'core.stdc.stdarg.va_copy' is not nothrow
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(321): Error: 'core.stdc.stdarg.va_end' is not nothrow
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(291): Error: function 'dmd.root.outbuffer.OutBuffer.vprintf' is nothrow yet may throw
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(340): Error: 'core.stdc.stdarg.va_start!(const(char)*).va_start' is not nothrow
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(342): Error: 'core.stdc.stdarg.va_end' is not nothrow
/home/vsts/work/1/s/parts/ldc-bootstrap/src/dmd/root/outbuffer.d(337): Error: function 'dmd.root.outbuffer.OutBuffer.printf' is nothrow yet may throw
ninja: build stopped: subcommand failed.
Failed to run 'override-build': Exit code was 1.
```

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -1,5 +1,5 @@
 name: ldc2
-version: 1.11.0
+version: 1.12.0
 summary: D compiler with LLVM backend
 description: |
     LDC is a portable compiler for the D programming Language, with
@@ -26,14 +26,13 @@ apps:
 parts:
   ldc:
     source: https://github.com/ldc-developers/ldc.git
-    source-tag: &ldc-version v1.11.0
+    source-tag: &ldc-version v1.12.0
     source-type: git
     plugin: cmake
     override-build: |
       cmake ../src \
         -DCMAKE_INSTALL_PREFIX= \
         -DD_COMPILER=../../ldc-bootstrap/build/bin/ldmd2 \
-        -DD_FLAGS='-w;-flto=thin' \
         -DCMAKE_BUILD_TYPE=Release \
         -DCMAKE_C_FLAGS_RELEASE='-O3 -DNDEBUG -Wa,-mrelax-relocations=no' \
         -DLIB_SUFFIX="$(getconf LONG_BIT)" \
@@ -90,16 +89,17 @@ parts:
     - -*
     build-packages:
     - g++
-    - ldc
     - libedit-dev
     - ninja-build
     - zlib1g-dev
+    build-snaps:
+    - ldc2/1.11/stable
     after:
     - llvm
 
   llvm:
     source: https://github.com/ldc-developers/llvm.git
-    source-tag: ldc-v6.0.1-3
+    source-tag: ldc-v7.0.1
     source-type: git
     plugin: cmake
     configflags:

